sqlengine/postgres_engine.go

```diff
@@ -501,8 +501,10 @@ const ensurePermissionsTriggersPattern = `
 	create or replace function make_readable_generic() returns void language plpgsql set search_path to public as {{.makeReadableGenericBodyStr}};
 	create or replace function make_readable() returns event_trigger language plpgsql set search_path to public as $$
 	begin
-		EXECUTE 'select make_readable_generic()';
-		RETURN;
+		IF NOT EXISTS (SELECT 1 FROM pg_event_trigger_ddl_commands() WHERE schema_name LIKE 'pg_temp%') THEN
```

Comment:
Surely you want `IF EXISTS` ... `schema_name NOT LIKE 'pg_temp%'` ?

In case there's a temp **and** a non-temp table.

Reply:
Also I take it you've looked and there really isn't a better way of detecting temp tables than this.

Reply:
Good suggestions on the logic

Yeah we can't do a filter in the trigger we checked.

Reply:
Although thinking about it more, this gets triggered on a table change, and I don't know of any commands that can alter multiple tables at once. But my SQL might be lacking.

Reply:
both versions behave the same for a single table change, so we did go with the safe option.